pom.xml

```diff
@@ -2651,6 +2651,7 @@
         <hadoop.version>3.1.0</hadoop.version>
         <curator.version>2.12.0</curator.version>
         <zookeeper.version>3.4.9</zookeeper.version>
+        <jetty.version>9.3.24.v20180605</jetty.version>
```

Comment:
Downgrade it, otherwise:
```
[info] YarnClusterSuite:
[info] org.apache.spark.deploy.yarn.YarnClusterSuite *** ABORTED *** (177 milliseconds)
[info]   org.apache.hadoop.yarn.exceptions.YarnRuntimeException: java.lang.NoSuchMethodError: org.eclipse.jetty.server.session.SessionHandler.getSessionManager()Lorg/eclipse/jetty/server/SessionManager;
```
Furthermore, We have some discuss about this change:
https://issues.apache.org/jira/browse/HADOOP-16152

Reply:
Hm, I'm afraid this might conflict with Java 11 compatibility, as this was updated for that reason: https://github.com/apache/spark/pull/22993
https://www.eclipse.org/lists/jetty-announce/msg00124.html

I wonder if this can be worked around on the Spark side, or whether it is indeed possible for Hadoop 3 to update?

Reply:
We shouldn't change Spark's version of Jetty because a YARN test is failing. A Spark application will not hit the same code paths as the YARN tests (which run a YARN server).

Maybe change the version of jetty in the YARN module when testing with hadoop-3, if there's no other way to work around this.

Reply:
I tried several ways but they all failed:
1. Use `hadoop-client-minicluster` instead of `hadoop-yarn-server-tests`: https://github.com/wangyum/spark-hadoop-client-minicluster
2. Change the version of jetty in the YARN module when testing with hadoop-3
3. Upgrade Eclipse Jetty version to 9.4.x for Hadoop: https://issues.apache.org/jira/browse/HADOOP-16152

Reply:
The ideal outcome is updating Hadoop, as otherwise it seems like Hadoop 3 and Java 11 support are in conflict. Then again, if Hadoop 3.x isn't quite going to work with Java 11 for other reasons, we have larger problems. Hadoop 2.x is currently working OK with Java 11 tests (it's Hive that's the issue) so I'm kind of surprised.

How did you try to override the Jetty version in the YARN module? I'd expect that's entirely possible

Reply:
I try to  override the Jetty version in the YARN module by: https://github.com/apache/spark/pull/24115/commits/29e583f59685f1343f618269b22c802f7ed553df.
But ```+-org.eclipse.jetty:jetty-servlet:9.3.24.v20180605 (evicted by: 9.4.12.v20180830)```.

I found that adapted `SessionHandler` from jetty-9.3.25.v20180904 can test the YARN module in Hadoop 2 and Hadoop 3. I know this is not a good way. But it seems that this is the only way at the moment.